src/ui/src/wds/WdsCopyClipboardButton.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+	<div ref="trigger" class="WdsCopyClipboardButton">
+		<WdsButton
+			variant="tertiary"
+			size="smallIcon"
+			class="WdsCopyClipboardButton__button"
+			:class="{
+				copied,
+			}"
+			:disabled="!isSupported"
+			:aria-label="
+				isSingleButtonMode
+					? label || 'Copy to clipboard'
+					: 'Open options'
+			"
+			:aria-haspopup="isSingleButtonMode ? undefined : 'menu'"
+			:aria-expanded="isSingleButtonMode ? undefined : String(isMenuOpen)"
+			:data-writer-tooltip="tooltipMessage"
+			data-writer-tooltip-placement="top"
+			@click="onCopyButtonClick"
+		>
+			<WdsIcon :name="copied ? 'check' : 'clipboard'" />
+		</WdsButton>
+		<WdsDropdownMenu
+			v-if="!isSingleButtonMode && isMenuOpen"
+			ref="menu"
+			class="WdsCopyClipboardButton__menu"
+			:options="options"
+			:style="floatingStyles"
+			@select="onMenuItemSelect"
+		/>
+	</div>
+</template>
+
+<script lang="ts">
+import type { WdsDropdownMenuOption } from "@/wds/WdsDropdownMenu.vue";
+
+export type WdsCopyClipboardButtonOption = Pick<
+	WdsDropdownMenuOption,
+	"label" | "value"
+>;
+</script>
+
+<script setup lang="ts">
+import { useClipboard } from "@vueuse/core";
+import WdsButton from "@/wds/WdsButton.vue";
+import WdsIcon from "@/wds/WdsIcon.vue";
+import WdsDropdownMenu from "@/wds/WdsDropdownMenu.vue";
+import { computed, nextTick, PropType, ref, useTemplateRef, watch } from "vue";
+import { autoPlacement, useFloating } from "@floating-ui/vue";
+import { useFocusWithin } from "@/composables/useFocusWithin";
+
+const props = defineProps({
+	label: { type: String, required: false, default: "" },
+	value: { type: String, required: false, default: "" },
+	options: {
+		type: Array as PropType<WdsCopyClipboardButtonOption[]>,
+		required: false,
+		default: () => [],
+	},
+});
+
+const { copy, copied, isSupported } = useClipboard();
+
+const isSingleButtonMode = computed(() => props.options.length === 0);
+
+const tooltipMessage = computed(() => {
+	if (isSingleButtonMode.value) {
+		return copied.value ? "Copied to clipboard" : props.label;
+	}
+
+	return undefined;
+});
+
+const isMenuOpen = ref(false);
+
+const trigger = useTemplateRef("trigger");
+const menu = useTemplateRef("menu");
+
+const { floatingStyles } = useFloating(trigger, menu, {
+	placement: "bottom-end",
+	middleware: [autoPlacement({ allowedPlacements: ["bottom-end"] })],
```

Comment:
Do we really need this middleware? I think it's only useful if you allow it to be placed on the top.

```suggestion
	middleware: [autoPlacement({ allowedPlacements: ["bottom-end", "top-end"] })],
```

or

```suggestion
```